TwinePlatform/twine-monolith
# ✨ Server-side sessions

fixes #207 

~80% of the diff is renaming and config changes.

Manually tested with visitor, volunteer and dashboard apps.

### Changes
- Use `@hapi/yar` and `@hapi/catbox-redis` to setup server-side session cache
- Create `session_cookie` authentication schema to validate requests using the session cache
- Re-write the `standard` authentication strategy to use the `session_cookie` schema and rely on `yar`
- Add config for the new auth system and the session cache
- Add a bit of documentation
- Update tests
- Remove pointless `src/routes` file
- Remove `hapi-auth-jwt2` and `jsonwebtoken`

### Testing Requirements

- [x] **Visitor app**

- Sign in
- Register visit
- Go to admin area
- Visit an admin page
- Sign out
- Try to (unsuccessfully) navigate directly to admin area (by typing in URL)


- Sign in
- Create log
- Update user info
- Sign out


- Sign in
- Click one of the tiles
- Check data loads
- Sign out

### Release Requirements
- Thursday 8th August

### Manual Deployment
- [ ] **Visitor app**
